ee/secureenclavesigner/secureenclavesigner.go

```diff
@@ -0,0 +1,197 @@
+//go:build darwin
+// +build darwin
+
+package secureenclavesigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/user"
+	"strings"
+	"time"
+
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+type SignRequest struct {
+	// Challenge is the B64 encoded challenged generated by the server
+	Challenge []byte `msgpack:"challenge"`
+	// Digest is the hash []byte of the data to be signed
+	Digest []byte `msgpack:"digest"`
+	// PubKey is the B64 encoded DER of the public key to be used to verify the signature
+	PubKey []byte `msgpack:"pub_key"`
+}
+
+type opt func(*secureEnclaveSigner)
+
+// WithExistingKey allows you to pass the public portion of an existing
+// secure enclave key to use for signing
+func WithExistingKey(publicKey *ecdsa.PublicKey) opt {
+	return func(ses *secureEnclaveSigner) {
+		ses.pubKey = publicKey
+	}
+}
+
+type secureEnclaveSigner struct {
+	uid                  string
+	challenge            []byte
+	pubKey               *ecdsa.PublicKey
+	pathToLauncherBinary string
+}
+
+func New(uid string, challenge []byte, opts ...opt) (*secureEnclaveSigner, error) {
```

Comment:
should we go ahead and create / verify the key exists in secure enclave in the new func? Fail faster at the expense of a possible extra exec?